src/commonTest/kotlin/exchange.dydx.abacus/app/manager/v2/V4TransactionTests.kt

```diff
@@ -177,6 +203,85 @@ class V4TransactionTests : NetworkTests() {
         assertTransactionQueueEmpty()
     }
 
+    @Test
+    fun testTriggerOrdersTransactionQueue() {
+        setStateMachineConnected(stateManager)
+        testWebSocket?.simulateReceived(mock.accountsChannel.v4_channel_data_with_orders)
+
+        val transactionQueue = subaccountSupervisor?.transactionQueue
+        var transactionCalledCount = 0
+        val transactionCallback: TransactionCallback = { _, _, _ -> transactionCalledCount++ }
+
+        val btcStopLossTriggerPrice = "30000"
+        val btcStopLossLimitPrice = "31000"
+        val btcTakeProfitTriggerPrice = "2000"
+        val btcTakeProfitLimitPrice = "2100"
+
+        val ethStopLimitOrderId = "31d7d484-8685-570c-aa62-c2589cb6c8d8"
+        val ethStopLimitOrderSize = "0.01"
+        val ethStopLimitTriggerPrice = "3300"
+        val ethStopLimitLimitPrice = "2100"
+
+        fun simulateNewBtcOrder(slTriggerPrice: String? = btcStopLossTriggerPrice, tpTriggerPrice: String? = btcTakeProfitTriggerPrice, slLimitPrice: String? = btcStopLossLimitPrice, tpLimitPrice: String? = btcTakeProfitLimitPrice) {
+            triggerOrdersInput("BTC-USD", slTriggerPrice, tpTriggerPrice, slLimitPrice, tpLimitPrice)
+            subaccountSupervisor?.commitTriggerOrders(0, transactionCallback)
+        }
+
+        fun simulateStopLimitOrderReplacement(triggerPrice: String? = ethStopLimitTriggerPrice, limitPrice: String? = ethStopLimitLimitPrice, size: String? = ethStopLimitOrderSize) {
+            triggerOrdersInput(marketId = "ETH-USD", stopLossTriggerPrice = triggerPrice, stopLossLimitPrice = limitPrice, stopLossOrderId = ethStopLimitOrderId, size = size)
+            subaccountSupervisor?.commitTriggerOrders(0, transactionCallback)
+        }
+
+        fun clearTransactions(numTimes: Int) {
+            repeat(numTimes) {
+                testChain?.simulateTransactionResponse(testChain!!.dummySuccess)
+            }
+            assertEquals(0, transactionQueue?.size)
+        }
+
+        simulateNewBtcOrder(null, null, null, null)
+        assertTransactionQueueEmpty("No orders enqueued")
+        assertEquals(1, transactionCalledCount) // Signals to FE that actions have been processed
+
+        simulateNewBtcOrder(tpTriggerPrice = null, slLimitPrice = null, tpLimitPrice = null)
+        assertTransactionQueueStarted()
+
+        // Creating New Orders
+        simulateNewBtcOrder(slLimitPrice = null, tpLimitPrice = null) // 2 new market orders created
+        assertEquals(2, transactionQueue?.size)
+        assertEquals(3, transactionCalledCount)
+        clearTransactions(2)
+
+        simulateNewBtcOrder() // 2 new limit orders created
+        assertEquals(2, transactionQueue?.size)
+        clearTransactions(2)
+
+        // Updating Existing Order
+        // testWebSocket?.simulateReceived(mock.accountsChannel.v4_subaccounts_update_1)
```

Comment:
hmm I having trouble making this set of tests work @johnqh because I think of how subaccounts are handled differently in v2 D: `testWebSocket?.simulateReceived(mock.accountsChannel.v4_subaccounts_update_1)` doesn't seem to quite be applying. Do you happen to know what the best way of getting this new position mocked in would be for the test?